server.js

```diff
@@ -291,11 +291,18 @@ const getImageFromMessage = async (message) => {
   if (message.document?.thumb) {
     return await getImageUrlFromPhotoSize(message.document.thumb);
   }
+  if (message.link_preview_options?.url) {
+    return message.link_preview_options?.url;
+  }
   if (message.entities && message.text) {
-    const urlEntity = message.entities.find((entity) => entity.type === "url");
-    return urlEntity
-      ? message.text.substring(urlEntity.offset, urlEntity.offset + urlEntity.length)
-      : false;
+    message.entities.forEach((entity) => {
```

Comment:
would you rewrite this loop with `for...of` ? it would make the return statement less confusing.